assets/js/base/context/providers/cart-checkout/checkout-processor.ts

```diff
@@ -213,97 +219,67 @@ const CheckoutProcessor = () => {
 			  }
 			: {};
 
-		const data = {
-			billing_address: emptyHiddenAddressFields(
-				currentBillingAddress.current
-			),
-			customer_note: orderNotes,
-			create_account: shouldCreateAccount,
-			...paymentData,
-			extensions: { ...extensionData },
-		};
-
-		if ( cartNeedsShipping ) {
-			data.shipping_address = emptyHiddenAddressFields(
-				currentShippingAddress.current
-			);
-		}
-
 		triggerFetch( {
 			path: '/wc/store/v1/checkout',
 			method: 'POST',
-			data,
+			data: {
+				shipping_address: cartNeedsShipping
+					? emptyHiddenAddressFields( currentShippingAddress.current )
+					: undefined,
+				billing_address: emptyHiddenAddressFields(
+					currentBillingAddress.current
+				),
+				customer_note: orderNotes,
+				create_account: shouldCreateAccount,
+				...paymentData,
+				extensions: { ...extensionData },
+			},
 			cache: 'no-store',
 			parse: false,
 		} )
-			.then( ( response ) => {
+			.then( ( response: unknown ) => {
+				assertResponseIsValid( response );
```

Comment:
If you accept the changes I suggested to make `assertResponseIsValid` generic you could change this to 

```suggestion
				assertResponseIsValid< CheckoutResponseSuccess >( response );
```

and remove the `as Promise< CheckoutResponseSuccess >;` below